packages/react-native-bridge/ios/RNReactNativeGutenbergBridge.swift

```diff
@@ -421,11 +421,23 @@ public class RNReactNativeGutenbergBridge: RCTEventEmitter {
     func toggleRedoButton(_ isDisabled: Bool) {
         self.delegate?.gutenbergDidRequestToggleRedoButton(isDisabled)
     }
+    
+    @objc
+    func requestConnectionStatus(_ callback: @escaping RCTResponseSenderBlock) {
+        callback([self.delegate?.gutenbergDidRequestConnectionStatus() ?? true])
+    }
 
-	@objc
-	func requestConnectionStatus(_ callback: @escaping RCTResponseSenderBlock) {
-		callback([self.delegate?.gutenbergDidRequestConnectionStatus() ?? true])
-	}
+    @objc
+    func logException(_ rawException: [AnyHashable: Any], callback: @escaping RCTResponseSenderBlock) {
+        guard let exception = GutenbergJSException(from: rawException) else {
+            callback([false])
+            return
+        }
+    
+        self.delegate?.gutenbergDidRequestLogException(exception) {
+            callback([true])
```

Comment:
Considering the argument (a.k.a the `wasSent` argument in the JS function) is always true, what do you think removing it instead?

In reality, we can't really know if an even is actually being sent. We know it's received by Sentry SDK, but it's totally up to Sentry when it comes to actually sending the event to the server.

Reply:
It's true that `wasSent` won't tell if the event was received by Sentry. However, this parameter can be `false` if the `rawException` argument is malformed and can't be parsed to a `GutenbergJSException` and hance won't be sent to the Crash logging service. Maybe we should rename it to clarify the purpose of the parameter to `wasProcessed` or similar, WDYT?

https://github.com/WordPress/gutenberg/blob/c017e122aa58dbfcc5e33fad72a1b9207ac77880/packages/react-native-bridge/ios/RNReactNativeGutenbergBridge.swift#L432-L435

Reply:
Or maybe we could consider that `wasSent` is related to sending the exception to the Crash logging service, not specifically that was actually sent and processed by Sentry. WDYT?

Reply:
I must have missed the `callback([false]) ` line above. The current code looks good to me, because there are different argument value used. 👍

Reply:
I'm not so sure we need to inform the JS side that the rawException is malformed. Right now, all it does is log the error, which seems we could do here in the bridge code.

I can see using a callback if the RN component will change given something failing down the call stack. The boundary component is already handling the error gracefully; everything else feels like "call and move on", much like how we call into the Sentry SDK without really knowing what happens next.

Reply:
I should note that the callback feels like a lot of boilerplate on Android. It would be simpler to log errors at the point of error vs bubbling that information back up to the RN boundary component.

Reply:
I agree that the parameter returned in the callback could be dropped. However, we need the callback to ensure that unhandled exceptions are logged. Following [this code](https://github.com/wordpress-mobile/gutenberg-mobile/blob/a88a07389f55bcf5a233a57ff01802dbe3968a04/src/errorLogging/index.js#L35-L42), if we call `defaultHandler` right after calling the bridge function `logException`, the exception is not logged because the undhandled exception leads to crashing the app before the Sentry event is sent. This is the main reason why I introduced the callback.

Reply:
> Following [this code](https://github.com/wordpress-mobile/gutenberg-mobile/blob/a88a07389f55bcf5a233a57ff01802dbe3968a04/src/errorLogging/index.js#L35-L42), if we call defaultHandler right after calling the bridge function logException,

Ah, okay. Thanks for pointing that out. I wasn't sure where the `defaultHandler` came from, but I see it's a ReactNative global, so we need to intercept that.  Makes sense now.

Reply:
> the undhandled exception leads to crashing the app

@fluiddot By "crashing the app", do you mean the iOS app, or the Gutenberg Editor? Out of curiosity, if an unexpected JS exception happens, it'd only affect the running JavaScriptCore VM, not the host app, right?

Reply:
The iOS app. By default, React Native causes a native crash for unhandled exceptions like other exceptions that might happen in the app.

In the future, we could change this behavior to closing the editor (i.e. a soft crash of the editor) instead of crashing and present post recovery options.

Reply:
Thanks for the explanation!